core/src/main/resources/META-INF/native-image/com.datastax.oss/java-driver-core/native-image.properties

```diff
@@ -0,0 +1 @@
+Args = -H:IncludeResources=.*reference\\.conf -H:IncludeResources=.*Driver\\.properties -H:DynamicProxyConfigurationResources=${.}/proxy.json -H:ReflectionConfigurationResources=${.}/reflection.json
```

Comment:
Just for my own education on GraalVM tooling: when this file is included in a jar, does it get automatically picked by `native-image`? What happens if a user doesn't want to use the default proxy.json and reflection.json files?

Reply:
Also, it just occurred to me that we have a Service implementation in driver-core: `DriverBlockHoundIntegration`. This impl is meant to be discovered via the ServiceLoader API. Therefore I think we need to:

1. Declare that implementation in reflection.json: the FQCN is `com.datastax.oss.driver.internal.core.util.concurrent.DriverBlockHoundIntegration`;
2. Also add `-H:IncludeResources=META-INF/services/*.*` to the arg line, in order to preserve the link between service interface and service impl.

Reply:
Re: usage of this file by native-image... yes, it's discovered as part of the build process.  https://www.graalvm.org/reference-manual/native-image/BuildConfiguration/ has a bit more detail.  Under this scheme there isn't a clean way for the user to avoid the specified reflection + proxy configs... so the defaults will always be included in the built native image even if they select custom impls of some of the components.  In the medium- to long-term this could be ameliorated by switching to a Feature impl.

Reply:
Re: the ServiceLoader configs... I haven't explicitly tested this but based on https://github.com/oracle/graal/issues/563 it looks like we should be okay